code/modules/mob/inventory.dm

```diff
@@ -609,3 +596,17 @@
 	for (var/obj/item/implant/storage/internal_bag in implants)
 		belongings += internal_bag.contents
 	return belongings
+
+/// Safely drop everything, without deconstructing the mob
+/mob/living/proc/drop_everything(del_on_drop, force, del_if_nodrop)
+	. = list() //list of items that were successfully dropped
+
+	var/list/all_gear = get_all_gear(recursive = FALSE)
+	for(var/obj/item/item in all_gear)
+		if(dropItemToGround(item, force))
+			if(del_on_drop)
+				qdel(item)
+				continue
+			. += item
```

Comment:
Wanna ensure that its not QDELETED, due to DROPDEL and alike

Reply:
Ah